### DISCUSSION TITLE
assignment 7th 관련 질문 - 버그

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#85] [Q&A] [answered]
코드를 지우고 커밋해버려서 기록은 안남았지만 코드와 상황을 설명해보겠습니다.

operator new를 대충 인터넷에 검색해서 나오는 코드로 넣어 보았습니다.
```
void * operator new (size_t size)
{
  void * adr = new char[size];
  return adr;
}

```

printmem이 무한이 호출된것같았습니다(아마도?)

이래서 중간중간 커밋하라는 것인가봅니다.

ㅠㅠ

**Comment by Woon-2** [comment, marked as answer]
`printmem`이 계속 `call`되는 것이 아닙니다.
`printmem`은 `operator new`를 이용하지도 않는데 어떻게 그런 일이 가능한가요...

첨부하신 `code`는 `operator new`를 `define`하는 도중에 `operator new`를 쓰고 있지 않습니까.   
(엄밀히는 `new expression` 속에서 `operator new`가 불릴 겁니다.)   
`recursion(재귀)`가 일어날 것이 예상되네요.   

`operator new`를 이용하는 순간 `process`는 `infinite loop`에 빠질 겁니다.

**Comment by Lcat12345** [comment]
new를 쓰면 안됫었군요.
